model/src/main/proto/performance_metrics.proto

```diff
@@ -0,0 +1,149 @@
+syntax = "proto3";
+
+package model;
+
+option java_package = "org.oppia.android.app.model";
+option java_multiple_files = true;
+
+// Structure for an oppia metric log.
+message OppiaMetricLog {
+  // The timestamp of the event.
+  // unit: long/millis, timezone: UTC
+  int64 timestamp_millis = 1;
+
+  // The priority of the event.
+  Priority priority = 2;
+
+  // Indicates whether the application is in foreground.
+  bool is_app_in_foreground = 3;
+
+  // The memory tier of the current device.
+  MemoryTier memory_tier = 4;
+
+  // The storage tier of the current device.
+  StorageTier storage_tier = 5;
+
+  // The current network type of the device.
+  NetworkType network_type = 6;
+
+  // The current activity where the event is getting logged.
+  CurrentActivity current_activity = 7;
+
+  // Structure of the apk size package.
+  message ApkSize {
+    // Size of the android package file of the app on the device in bytes.
+    int64 apk_size_bytes = 1;
+  }
+
+  // Structure of the storage usage package.
+  message StorageUsage {
+    // Amount of storage usage by the app on user's device in bytes.
+    int64 storage_usage_bytes = 1;
+  }
+
+  // Structure of the startup latency package.
+  message StartupLatency {
+    // Amount of latency on application startup in millis.
+    int64 startup_latency_millis = 1;
+  }
+
+  // Structure of the memory usage package.
+  message MemoryUsage {
+    // Amount of memory used by the application on the device in bytes.
+    int64 total_pss_bytes = 1;
+  }
+
+  // Structure of the network usage package.
+  message NetworkUsage{
+    // Number of bytes received by the application on the device during the bucket's time interval.
+    int64 bytes_received = 1;
+
+    // Number of bytes transmitted by the application from the device during the bucket's time
+    // interval.
+    int64 bytes_transmitted = 2;
+  }
+
+  // Structure of the cpu usage package.
+  message CpuUsage{
+    // Amount of cpu used by the application.
+    int64 cpu_usage = 1;
+  }
+
+  // Supported priority of events for performance metric logging.
+  enum Priority {
+    // The undefined priority of an event.
+    PRIORITY_UNSPECIFIED = 0;
+
+    // The priority of events whose logs will be the first to be removed from the storage if the
+    // size exceeds a certain limit
+    LOW_PRIORITY = 1;
+
+    // The priority of events whose logs will be the ones to be removed from the storage if the
+    // size exceeds a certain limit and no low priority logs exist.
+    MEDIUM_PRIORITY = 2;
+
+    // The priority of events whose logs should not be removed from the storage on most occasions
+    // but can be removed if they're the only ones there and size limit exceeds.
+    HIGH_PRIORITY = 3;
+  }
+
+  // Supported activities for indicating the current activity where the event logging is occuring.
+  enum CurrentActivity {
```

Comment:
Perhaps instead this should be 'screen' instead of 'Activity'? The latter would require tracing to implement generically, or we'll have to remember to update this enum (which is a bit error-prone). We could go with the tracing approach in the future, but for now maybe we can just abstract it to screens (even though generally screens->activities will be a 1:1 relationship, the benefit is not needing to necessarily know the activity at the domain level, just which screen it corresponds to).